payments/stripe/__init__.py

```diff
@@ -76,3 +90,146 @@ class StripeCardProvider(StripeProvider):
+    product_data: StripeProductData
+    unit_amount: int
+    recurring: Optional[dict] = field(init=False, repr=False, default=None)
+    tax_behavior: Optional[str] = field(init=False, repr=False, default=None)
+
+
+@dataclass
+class StripeLineItem:
+    price_data: StripePriceData
+    quantity: int
+    adjustable_quantity: Optional[dict] = field(init=False, repr=False, default=None)
+    dynamic_tax_rates: Optional[dict] = field(init=False, repr=False, default=None)
+    tax_rates: Optional[str] = field(init=False, repr=False, default=None)
+
+
+class StripeProviderV3(BasicProvider):
+    """Provider backend using `Stripe <https://stripe.com/>` api version 3_.
+
+    This backend does not support fraud detection.
+
+    :param secret_key: Secret key assigned by Stripe.
+    :param payment_method_types: From Stripe API, comma separated
+    :param use_token: Use instance.token instead of instance.pk in client_reference_id
+    """
+
+    form_class = PaymentFormV3
+    session = None
+
+    def __init__(
+        self,
+        secret_key,
+        payment_method_types="card",
+        use_token=True,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.secret_key = secret_key
+        self.payment_method_types = payment_method_types.split(",")
+        self.use_token = use_token
+
+    def get_form(self, payment, data=None):
+        if not payment.transaction_id:
+            try:
+                self.create_session(payment)
+            except PaymentError as pe:
+                payment.change_status(PaymentStatus.ERROR, str(pe))
+                raise PaymentError(pe)
+            else:
+                if not self.session:
+                    raise PaymentError(_("self.session can't be None"))
+                payment.attrs.session = self.session
+                payment.transaction_id = self.session.get("id", None)
+                payment.save()
+
+        if "url" not in payment.attrs.session:
+            raise PaymentError(_("No `url` in payment.attrs.session"))
+
+        raise RedirectNeeded(payment.attrs.session.get("url"))
+
+    def create_session(self, payment):
+        """Makes the call to Stripe to create the Checkout Session"""
+        if not payment.transaction_id:
+            stripe.api_key = self.secret_key
+            session_data = {
+                "payment_method_types": self.payment_method_types,
+                "line_items": self.get_line_items(payment),
+                "mode": "payment",
+                "success_url": payment.get_success_url(),
+                "cancel_url": payment.get_failure_url(),
+                "client_reference_id": payment.token if self.use_token else payment.pk,
+            }
+            # Patch session with billing email if exists
+            if payment.billing_email:
+                session_data.update({"customer_email": payment.billing_email})
+            try:
+                self.session = stripe.checkout.Session.create(**session_data)
```

Comment:
Why not `return` the session instead? This would make `get_form` a bit simpler too (since you don't need to check whether it is `None`).